content/guides/reader_conditionals.adoc

```diff
@@ -0,0 +1,198 @@
+
+[source,clojure]
+----
+(ns route-ccrs.schema.ids.part-no-test
+  (:require #?(:clj  [clojure.test :refer :all]
+               :cljs [cljs.test :refer-macros [is]])
+            #?(:cljs [cljs.test.check :refer [quick-check]])
+            #?(:clj  [clojure.test.check.clojure-test :refer [defspec]]
+               :cljs [cljs.test.check.cljs-test :refer-macros [defspec]])
+            #?(:clj  [clojure.test.check.properties :as prop]
+               :cljs [cljs.test.check.properties :as prop
+                       :include-macros true])
+            [schema.core :as schema :refer [check]]
+            [route-ccrs.schema.ids :refer [PartNo]]
+            [route-ccrs.generators.part-no
+             :refer [gen-part-no gen-invalid-part-no]]))
+----
+
+In this example, we want to be able to use the `rethinkdb.query` namespace in Clojure and ClojureScript. However we can't load the required `rethinkdb.net` in ClojureScript as it uses Java sockets to communicate with the database. Instead we use a reader conditional so it's only required when read by Clojure programs.
+
+[source,clojure]
+----
+(ns rethinkdb.query
+  (:require [clojure.walk :refer [postwalk postwalk-replace]]
+            [rethinkdb.query-builder :as qb :refer [term parse-term]]
+            #?(:clj [rethinkdb.net :as net])))
+
+;; snip...
+
+#?(:clj (defn run [query conn]
+      (let [token (get-token conn)]
+        (net/send-start-query conn token (replace-vars query)))))
+----
+
+== Exception handling
+
+Exception handling is another area that benefits from reader conditionals. ClojureScript supports `(catch :default)` to catch everything and this may come Clojure too, however you will often still want to handle host specific exceptions. Here's an https://github.com/runexec/lemon-disc/blob/c24c6638f1d476a0f5470387e52a2b702117c4a9/src/lemon_disc/core.cljc#L65-L72[example] from https://github.com/runexec/lemon-disc[lemon-disc].
+
+[source,clojure]
+----
+(defn message-container-test [f]
+  (fn [mc]
+      (passed?
+        (let [failed* (failed mc)]
+          (try
+            (let [x (:data mc)]
+              (if (f x) mc failed*))
+            (catch #?(:clj Exception :cljs js/Object) _ failed*))))))
+----
+
+== Splicing
+
+The splicing reader conditional is not as widely used as the standard one. For an example on its usage, let's get really meta and look at the https://github.com/clojure/clojure-clr/blob/544e9354e121e10a656702222d47c8398468fb02/Clojure/Clojure.Tests/clojure/test_clojure/reader.cljc#L672-L677[tests] for reader conditionals in the ClojureCLR reader. What might not be obvious at first glance is that the vectors inside the splicing reader conditional are being wrapped by a surrounding vector.
```

Comment:
"let's get really meta" - maybe a bit too informal? Also the section may bit a bit too short. Although the unit test clarifies details of splicing, it's not a good showcase of how or why the conditional is used in .cljc programs.